### PR TITLE
starbuck main stage

### PR DESCRIPTION
**Pull request by RamiroND**

### CODE REVIEW

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: 'Supabase Auth: Identity Linking, Session Control, Passwords and Hooks'
```
Comment (kiwicopple) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:2]:
```suggestion
title: 'Supabase Auth: Identity Linking, Hooks, and Have I Been Pwned integration'
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+description: Supacharging Supabase Auth with Identity Linking, Session Control, Passwords, and Hooks
```
Comment (kiwicopple) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:3]:
```suggestion
description: Four major Auth features: Identity Linking, Session Control, Leaked Password Protection, and Hooks
```

(let's avoid using `supa` wherever possible)

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  - launch-week
+  - auth
+date: '2023-12-14'
+toc_depth: 3
+author: stojan,joel,kangmingtay
+image: lwx-supabase-auth/auth-og.png
+thumb: lwx-supabase-auth/auth-thumb.png
+---
+
+Welcome to Launch Week X Day 4! We're excited to announce four new features for Supabase Auth:
```
Comment (kiwicopple) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:17]:
```suggestion
We're excited to announce four new features for Supabase Auth:
```

**Review by kiwicopple — COMMENTED** [review]
a few suggestions

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 4]
```diff
@@ -0,0 +1,192 @@
+---
+title: 'Supabase Auth: Identity Linking, Hooks, and Have I Been Pwned integration'
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:2]:
```suggestion
title: 'Supabase Auth: Identity Linking, Hooks, and HaveIBeenPwned integration'
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    title="YouTube video player"
+    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+    allowfullscreen
+  />
+</div>
+> Watch a demonstration of some of the new features
+
+## Identity Linking
+
+When a user signs in, an identity is created with the method and sign-in provider. Historically, [Supabase Auth](https://supabase.com/docs/guides/auth) has been automatically linking identities to a user if the identity shares the same verified email as the user. This is convenient to de-duplicate user accounts. However, some developers also need the flexibility to link accounts that don’t share the same email.
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:37]:
```suggestion
When a user signs in, an identity is created with the authentication method and sign-in provider. Historically, [Supabase Auth](https://supabase.com/docs/guides/auth) has been automatically linking identities to a user if the identity shares the same verified email as the user. This is convenient to de-duplicate user accounts. However, some developers also need the flexibility to link accounts that don’t share the same email. 
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+These session control settings are available on the Pro plan and above.
+
+IMAGE GOES HERE
+
+> For more information, see the [Session Management docs](https://supabase.com/docs/guides/auth/sessions).
+
+## Leaked Password Protection
+
+Passwords often lack security due to user habits like weak choices or reuse across several sites.
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:99]: hmm the phrasing sounds strange to me, i prefer:
```suggestion
Passwords can be inherently insecure due to common user behaviors like choosing guessable passwords or reusing them across different platforms. 
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  return jsonb_set(
+		event, 
+		'{claims,level}', 
+		user_level);
+
+end;
+$$
+```
+
+Then once you’ve created the function in the database, you only need to register it with Supabase Auth:
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:155]:
```suggestion
Once you’ve created the function in the database, you only need to register it with Supabase Auth:
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+IMAGE GOES HERE
+
+Initially, you can register an Auth Hook for the following points in the flow:
+
+- **Custom access token:** called each time a new JWT is generated.
+- **MFA verification attempt**: called each time an MFA factor is verified, allowing you finer control over detecting and blocking attempts.
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:162]:
```suggestion
- **MFA verification attempt**: called each time an MFA factor is verified, allowing finer control over detecting and blocking attempts.
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+- **Password verification attempt**: called each time a password is used to sign-in a user, allowing you finer control over the security of the user’s accounts.
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:163]:
```suggestion
- **Password verification attempt**: called each time a password is used to sign-in a user, allowing finer control over the security of the user’s accounts.
```

File: apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+Auth Hooks is available today for self-hosting and will be rolling out to the platform next month. Reach out to us via [support](https://supabase.help) if you need access sooner!
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-dentity-linking-session-control.mdx:165]:
```suggestion
Auth Hooks is available today for self-hosting and will be rolled out to the platform next month. Reach out to us via [support](https://supabase.help) if you need access sooner!
```

File: apps/www/_blog/2023-12-14-supabase-auth-identity-linking-hooks.mdx [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+const googleIdentity = identities
+	.find(({ provider }) => provider === 'google')
+
+// unlink the google identity from the user
+const { data, error } = await supabase
+	.auth
+	.unlinkIdentity(googleIdentity)
+```
+
+Currently, these methods support linking an OAuth identity. To link an email or phone identity to the user, you can use the `[updateUser()](https://supabase.com/docs/reference/javascript/auth-updateuser)`method.
```
Comment (kangmingtay) [inline apps/www/_blog/2023-12-14-supabase-auth-identity-linking-hooks.mdx:69]:
```suggestion
Currently, these methods support linking an OAuth identity. To link an email or phone identity to the user, you can use the [`updateUser()`](https://supabase.com/docs/reference/javascript/auth-updateuser)method.
```

**Pull request merged**